eventuals/collect.h

```diff
@@ -5,45 +5,70 @@
 #include "eventuals/loop.h"
 #include "eventuals/type-traits.h"
 
-// NOTE: The following include is a rather large one.
-#include "google/protobuf/repeated_field.h"
-
 ////////////////////////////////////////////////////////////////////////
 
 namespace eventuals {
 
 ////////////////////////////////////////////////////////////////////////
 
-template <typename Container>
+template <typename Collection, typename Enable = void>
+struct EventualsCollector {
+  template <typename T>
+  void Collect(Collection& collection, T&& value) {
+    static_assert(
+        !std::is_same_v<void, Enable>,
```

Comment:
Hmm, I'm not sure if this is doing what you want? I love the idea of a static time check here though, and I've been meaning to add an `always_false_v<T>` for a while. Let's add it now!

In `eventuals/type-traits.h` lets add:

```cpp
////////////////////////////////////////////////////////////////////////

template <typename>
inline constexpr bool always_false_v = false;

////////////////////////////////////////////////////////////////////////
``` 

Then here we can do:
```suggestion
    static_assert(
        always_false_v<T>,
```

Which should only get triggered if the compiler needs to compile this implementation!

Reply:
Hmm, I've played around with my implementation and it seems like it's working, but I do like your solution instead! I'll try it